esmvaltool/diag_scripts/ocean/diagnostic_tools.py

```diff
@@ -117,29 +190,67 @@ def add_legend_outside_right(plot_details, ax1, column_width=0.1):
         'label': label for the legend.
     ax1 is the axis where the plot was drawn.
     """
-    #####
+    # ####
     # Create dummy axes:
     legend_size = len(plot_details.keys()) + 1
-    ncols = int(legend_size / 25) + 1
     box = ax1.get_position()
-    ax1.set_position(
-        [box.x0, box.y0, box.width * (1. - column_width * ncols), box.height])
+    if loc.lower() == 'right':
+        nrows = 25
+        ncols = int(legend_size / nrows) + 1
+        ax1.set_position([box.x0,
+                          box.y0,
+                          box.width * (1. - column_width * ncols),
+                          box.height])
+
+    if loc.lower() == 'below':
+        ncols = 4
+        nrows = int(legend_size / ncols) + 1
+        ax1.set_position([box.x0,
+                          box.y0 + (nrows * column_width),
+                          box.width,
+                          box.height - (nrows * column_width)])
 
     # Add emply plots to dummy axis.
     for index in sorted(plot_details.keys()):
+        try:
+            colour = plot_details[index]['c']
+        except AttributeError:
```

Comment:
It seems unlikely that `plot_details[index]['c']` will raise an `AttributeError`, unless `plot_details` is not a dict of dicts as specified in the documentation above. Maybe you mean `KeyError`? In that case, you could also write:
```python
colour = plot_details[index].get('c', plot_details[index]['colour'])
```

Reply:
I've swapped out the 'c' or 'color' flags, swapped the `AttributeError` for `KeyError`. I've also changed the documentation to reflect the new flexibility.